### ISSUE TITLE
Way to load additional css files with hash in filename after app init.

### ISSUE THREAD

**Issue by SvenBudak** [angular/angular#35027]
I am working on a good solution for dark / light theming depends on `prefers-color-scheme`. This is what i got: https://medium.com/@svenbudak/how-to-implement-dark-light-mode-in-angular-mateiral-with-prefers-color-scheme-ce3e980e2ea5

The problem in this solution is that both themes are includet inside the app. Which can be cool, because if there are just some some little lines they overwrite some basic colors, there will be no loading time while switching theme. But if there are some more lines of theming css, its overhead to load both.

So i added to angular.json:
```
"styles": [
    "src/scss/styles.scss",
    {
        "input": "src/scss/prefers-color-scheme/dark.scss",
        "bundleName": "dark"
    },
    {
        "input": "src/scss/prefers-color-scheme/light.scss",
        "bundleName": "light"
    }
]
```

Which is working very well. It generates me this 2 files:
```
dark.4dec7b0304c3dec40370.css
light.33e2a5c89c5f95e86528.css
```

I also asked in the `@angular/components` issue [board](https://github.com/angular/components/issues/16905), but i think this is a angular task.

I know i can add in the angular.json the `inject` attribute, then they have no hashes and i can load them easy with renderer2. But then the PWA support will gets lost. They are any plans to make it possible to get somehow filenames like this after the app gets initialized? Because i need to check in app first the `prefers-color-scheme` (i will make this check in js file, because i do some more then only read it from browser).

**Comment by tomayac** [comment]
Would dynamically querying the file names help?

```js
const darkCSS = document.querySelector('link[rel="stylesheet"][href^="dark"]').href;
```

**Comment by SvenBudak** [comment]
I don't think so. Check the service I got. It's important to give people the possibility to overwrite the OS specific selected prefered-color-scheme. Input it into the localStorage. 

We need to check the result of the service and have to load depends on it the light or dark one. (Which filename we don't have)
